PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt

```diff
@@ -96,11 +127,28 @@ class PayPalWebCheckoutClient internal constructor(
 
     internal fun handleBrowserSwitchResult() {
         browserSwitchResult = browserSwitchClient.deliverResult(activity)
-        listener?.also {
-            browserSwitchResult?.also { result ->
-                when (result.status) {
-                    BrowserSwitchStatus.SUCCESS -> deliverSuccess()
-                    BrowserSwitchStatus.CANCELED -> deliverCancellation()
+        if (listener == null && vaultListener == null) {
```

Comment:
Would it be better to just use `listener?.let { /* invoke listener */ }` whenever we invoke the listeners throughout? Is there a chance with this logic that they've set the `listener` but not the `vaultListener` and would get an NPE if we try to invoke the `vaultLlistener` later on?

Reply:
Yeah it seems like the `listener?.also` pattern was there to skip parsing when no listener exists. I'm willing to bet though that this code doesn't work as intended. `deliverResult()` should only happen once, but the unit test mocks the behavior so that it does something different. We may need to refactor this because if I'm thinking it through correctly, the way [it's implemented currently](https://github.com/paypal/paypal-android/blob/fe8d82a0174e4bf49aaa2af5b8acd91bf428d459/PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt#L98) won't have a chance to defer delivery of a browser switch result.

When `browserSwitchClient.deliverResult(activity)` is called twice, the value at `browserSwitchResult` will become `null`. So even if a listener is set later on, there won't be a browser switch result to deliver because the pending result will have been cleared internally.

PPCP should use the `parseResult()` method as well. It'd be awesome if we could use an Activity launcher too because then browser switch could theoretically be launched from a Jetpack Composable without a direct activity reference.